blazarclient/command.py

```diff
@@ -329,3 +329,42 @@ def get_data(self, parsed_args):
         data = resource_manager.get_allocation(parsed_args.id)
         self.format_output_data(data)
         return list(zip(*sorted(data.items())))
+
+
+class ReallocateCommand(BlazarCommand):
+    """Reallocate host from current leases."""
+
+    api = 'reservation'
+    resource = None
+    log = None
+
+    def get_parser(self, prog_name):
+        parser = super(ReallocateCommand, self).get_parser(prog_name)
+        if self.allow_names:
+            help_str = 'ID or name of %s to update'
+        else:
+            help_str = 'ID of %s to update'
+        parser.add_argument(
+            'id', metavar=self.resource.upper(),
+            help=help_str % self.resource
+        )
+        self.add_known_arguments(parser)
+        return parser
+
+    def run(self, parsed_args):
+        self.log.debug('run(%s)' % parsed_args)
+        blazar_client = self.get_client()
+        body = self.args2body(parsed_args)
+        if self.allow_names:
+            res_id = utils.find_resource_id_by_name_or_id(blazar_client,
```

Comment:
I was stuck on this line until I realized this stood for "resource id", not "reservation id".

Reply:
Not totally sure either. Following conventions from Update.

I'll add to docs before merging.